src/preferences/dialog/dlgprefwaveform.h

```diff
@@ -5,6 +5,14 @@
 #include "preferences/dialog/dlgpreferencepage.h"
 #include "preferences/dialog/ui_dlgprefwaveformdlg.h"
 #include "preferences/usersettings.h"
+#include "waveform/widgets/waveformwidgettype.h"
+#ifdef MIXXX_USE_QOPENGL
+#include "waveform/renderers/allshader/waveformrenderersignalbase.h"
+#define DECL_SLOT_WAVEFORM_OPTION(opt)                                               \
+    void slotSetWaveformOption##opt(bool checked) {                                  \
+        slotSetWaveformOptions(allshader::WaveformRendererSignalBase::opt, checked); \
+    }
+#endif
```

Comment:
I also dislike macros, they obfuscate and make it harder to see what's happening.  sometimes boilerplate is boilerplate.  (But I will also not block on this)

Reply:
also this is only used a few times so it doesn't seem that helpful

Reply:
Yeah I don't mind just transforming to code. I just thought, as we discussed on the sync, we might want to add more option in the near future, and wanted to introduce some kinda establish way to handle option, like [done here for metadata](https://github.com/mixxxdj/mixxx/blob/6f3aea012ab6dee632b3fa1311f522e3b9a5bcb6/src/track/trackrecord.h#L36)

> I won't insist. Just sharing my opinion

Honestly don't mind either way :) I was just confused because sometime, that seem to be the preferred way, including on some recently merged code. 
I guess it would be nice to update the [wiki guideline](https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines) about macros, so it is clearer to me when and when to use it.

Reply:
good idea, and I see what you mean about it being used elsewhere.  A couple things I would say:
1. mixxx is super inconsistent in style, despite our best efforts.  Between the evolution of c++ over the past 20 years and the large number of disparate contributors, things are quite messy. Sorry!
2. This macro to me falls into YAGNI ("you ain't gonna need it") territory.  If we get to the point where there are a ton of options funcs and it gets overwhelming, then we can refactor and introduce the macro.  But I don't believe it makes sense to introduce it now with the assumption that we *might* or even *probably* will want it, because we may not :). Only write it when you actually do need it.

This comes down to "code smell" and personal preference, which is why comments like this are non-blocking. but imho, macros are very rarely worth it.